portfolio/src/main/webapp/script.js

```diff
@@ -13,16 +13,33 @@
 // limitations under the License.
 
 /**
+ To fetch & parse news articles from the NewsAPI
+    @param location name of the place we are asking the NewsApi about.
+ */
 
+async function fetchNews(location) {
+    await fetch(`/fetchNews?location=${location}`)
+        .then(response => response.json())
+        .then(articles => {
+            
+            const newsContainer = document.getElementById('news-container');
+            if(articles.status !== 'ok'){
+                newsContainer.innerText = "We were unable to find anything on this";
+            }else{   
+                console.log(articles.articles);
+                newsContainer.innerText = articles.articles[0].title;
+            }
+        })
```

Comment:
Remove this part so that the caller can determine what they want to do with the returned value.